Upgraded to MUI 5.4.3 (Latest) and React 17

Elius94 opened the pull request.

Elius94 committed:
Upgraded to MUI 5.4.3 (Latest) and React 17

Elius94 commented on LICENSE:
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2020 Andrew Rothbart
+Copyright (c) 2022 Elia Lazzari
```
Just replace with your name, it was an automatic snippet

rothbart commented on src/onepirate/modules/components/Snackbar.tsx:
```diff
@@ -42,7 +39,7 @@ const styles = (theme: Theme) =>
     },
   });
 
-function Transition(props: TransitionProps) {
```
Is there a strong type we can use here? Would prefer to avoid `any` if possible.

Elius94 replied:
I'm changing any with TransitionProps...
soon you will see the next commit.

rothbart commented on src/onepirate/modules/withRoot.tsx:
```diff
@@ -1,16 +1,42 @@
+
+
+
+declare module '@mui/styles/defaultTheme' {
```
What is this (and why are there three of them)?

Elius94 replied:
This is an error! I'm fixing it and I'll commit

rothbart reviewed (commented):
Thanks! A few small comments, looks great overall.

Elius94 commented:
Hi, thanks for reply. I've followed the migration guide from Material-ui 4 to the latest 5.x.x.
You can find that procedure in the official site: https://mui.com/guides/migration-v4/

In particular I've converted the import packages naming with the newer. 
The most impactful change is for the theming that was really different in v4 and so I've used an official automated script to convert v4 theming to v5. This is the reason of the strange rows of code that you have noticed.

Elius94 committed:
Fixed issues:
 - Multiple declare module
 - Slide transition in snackbar without type
 - React v17.0.2

Elius94 commented:
Ok, I've fixed the issues

rothbart commented:
Thank you for your PR! I merged it in here: https://github.com/rothbart/onepirate-typescript/pull/10

I noticed that there were a few changes to the typescript files for the onepirate theme (https://github.com/mui/material-ui/tree/master/docs/src/pages/premium-themes/onepirate) so I decided to just backport all the typescript changes :)

Thanks again for taking the initiative on this! Much appreciated.

Elius94 commented:
Good job! thanks
